store/src/java/com/zimbra/cs/service/account/ResetPassword.java

```diff
@@ -47,8 +48,13 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
         AuthProvider.validateAuthToken(prov, at, false);
 
         Account acct = at.getAccount();
+        boolean locked = acct.getBooleanAttr(Provisioning.A_zimbraPasswordLocked, false);
+        if (locked) {
+            throw AccountServiceException.PASSWORD_LOCKED();
+        }
         ResetPasswordUtil.validateFeatureResetPasswordStatus(acct);
         String newPassword = req.getPassword();
+        prov.checkPasswordStrength(acct, newPassword);
```

Comment:
instead of explicitly calling this method, could have just used prov.setPassword(acct, newPassword, true); where password is being set.